pkg/didcomm/dispatcher/api.go

```diff
@@ -78,14 +80,63 @@ type DIDCommAction struct {
 	Callback Callback
 }
 
+// Result is used for notifying whether callback was finished
+// the result is successful when Err is nil
+type Result struct {
+	Err error
+}
+
 // DIDCommCallback message type to pass service callback in go channels.
 type DIDCommCallback struct {
 	// Set the value in case of any error while processing the DIDComm message event by the consumer.
-	Err error
+	res  chan Result
+	done bool
+}
+
+// NewDIDCommCallback creates DIDCommCallback message
+func NewDIDCommCallback() *DIDCommCallback {
+	return &DIDCommCallback{}
+}
+
+// WithResult creates result chan which provides the possibility to be notified when the callback is finished
+func (c *DIDCommCallback) WithResult() *DIDCommCallback {
```

Comment:
Adding to the above point, I feel rather than giving the option to the consumer to receive the error result, service should always give back the error. It should be up-to the consumer to either process or ignore the error.

Reply:
Done!